### ISSUE TITLE
Add metrics for the status of NodeClaims?

### ISSUE THREAD

**Issue by diranged** [aws/karpenter-provider-aws#6802]
### Description

**What problem are you trying to solve?**

We discovered a NodeClaim in `status=unready` state - the Karpenter controller had been replaced at just the very moment a node came up, and it never applied the `karpenter.sh/initialized=true` label. That nodeclaim then just sat around ... unable to be cleaned up (I'll open a separate issue for this), and unnoticed.

I was surprised to not be able to find any metrics from Karpenter about the basic status of the NodeClaims. I'd like to see something like `karpenter_nodeclaim_status` that reports the status and conditions of nodeclaims, which would allow us to then alert our teams when something is wrong.

**How important is this feature to you?**

Without this - it will be hard to recommend we adopt Karpenter for more production workloads ...

**Comment by jigisha620** [comment]
Hi @diranged,
With Karpenter v1.0.0, we have [operator_status_condition_transitions_total](https://karpenter.sh/v1.0/reference/metrics/#operator_status_condition_transitions_total) that can be used to find the number of nodeClaims in a given status condition.

**Comment by diranged** [comment]
Hmm. We had a NodeClaim in an unready state for ~36 hours, but because this metric seems to be a _transition_ metric, it's really hard to utilize it to warn us when something like that happens. This feels more like it should be a gauge metric than a counter. Thoughts?

**Comment by jonathan-innis** [comment]
See [`operator_status_condition_count`](https://karpenter.sh/docs/reference/metrics/#operator_status_condition_count). This is our gauge representation of this metric.

**Comment by diranged** [comment]
@jonathan-innis Ah! Indeed, that metric reports what I am looking for... thank you. I can close this out. Quick note .. it would b great if the https://karpenter.sh/docs/reference/metrics page included what dimensions were reported on the metrics.

**Comment by jonathan-innis** [comment]
Yeah, I'd agree that we should have dimensions that are auto-gened in the docs -- particularly because the help text that we use quickly becomes out of date. We have this issue #5276 that's open to add this but it isn't something that we've had time to get around to. If that's something that you would be interested in adding, would love help on this!

**Comment by jonathan-innis** [comment]
Closing this out since we _do_ have the metrics right now -- let's add discussion on adding the dimensions on the linked issue!
